### PR TITLE
Fixes issue #137

### PR DESCRIPTION
**Pull request by Roshan-Thomas**
**Context:**
Unitary Hack task for updating the tutorial and documentation style.

**Description of the change:**
The following changes are implemented in this PR

- Upgraded to `sphinxawesome_theme` and made sure the changes were reflected in all the pages. 
- The white and dark logos were added for both the dark and light versions of the website.
- The `tutorial` was updated, and both required examples were added to it and deleted from the examples menu. 
- Updated the MBQC and LC-MBQC intros to reflect the requested paper in the issue. 

**Related issue:**
The related issue is #137.

### CODE REVIEW

**Comment by Roshan-Thomas** [comment]
Hey @shinich1, my apologies for getting back to you pretty late, I had a lot of uni work to do. 

I have completed the required changes,
- updated the code on the introduction and tutorial pages to reflect the latest changes to the codebase.
- made the tutorial page a lot easier to follow, and
- added graph visualizations on the tutorial page. 

The build can be found [here](https://app.readthedocs.org/projects/graphix-uh/builds/24601310/). I ran the build a while back and it still hasn't completed. It successfully installed graphix and then remain stuck there. 

Let me know if there are any more changes needed to be made.

**Comment by shinich1** [comment]
> Hey @shinich1, my apologies for getting back to you pretty late, I had a lot of uni work to do.
> 
> I have completed the required changes,
> 
> * updated the code on the introduction and tutorial pages to reflect the latest changes to the codebase.
> * made the tutorial page a lot easier to follow, and
> * added graph visualizations on the tutorial page.
> 
> The build can be found [here](https://app.readthedocs.org/projects/graphix-uh/builds/24601310/). I ran the build a while back and it still hasn't completed. It successfully installed graphix and then remain stuck there.
> 
> Let me know if there are any more changes needed to be made.

@Roshan-Thomas  Thanks! the docs seems to have built fine (it takes time building all the `examples` from the code). I have a few remaining requests before merging:
- can you make the logo on the top-left corner of the page larger, so that it's similar size as [current one ](https://graphix.readthedocs.io/en/latest/index.html). please remove the bold text next to the logo reading 'graphix documentation'.
- can you set up code highlighting across the docs? all code blocks are black, which is a little difficult to read.
- the heading level for 'simulating noisy mbqc', running pattern on quantum devices and generating qasm file need to be one level higher - see screenshot below, it currently looks as if they are part of 'optimizing patterns'
image

File: docs/source/tutorial.rst
Comment (shinich1) [inline docs/source/tutorial.rst:180]: can you change this line to 
`As we see below, we can simplify and optimize the pattern by calling various methods of `~graphix.pattern.Pattern` class.

File: docs/source/tutorial.rst
Comment (shinich1) [inline docs/source/tutorial.rst:222]: change this line to 
`The command sequence is now simpler and note that the most byproduct commands now apply to output nodes (3, 7).`

**Comment by Roshan-Thomas** [comment]
Hi @shinich1, I have added the following changes,

- removed 'graphix documentation' on top-left corner
- added code highlighting
- changed tutorial heading levels as per request
- updated the tutorial page to have simpler language as per request.

Note: The logo was something I was not able to enlarge it as the `spinxawesome-theme` does not have that feature. I have searched in the documentation and there is no mention on how to increase the size of the logo. If you know of the method to do it, please let me know and I can implement it.

File: docs/source/tutorial.rst
Comment (Roshan-Thomas) [reply docs/source/tutorial.rst]: added to PR

File: docs/source/tutorial.rst
Comment (Roshan-Thomas) [reply docs/source/tutorial.rst]: added to PR

**Comment by shinich1** [comment]
> Hi @shinich1, I have added the following changes,
> 
> * removed 'graphix documentation' on top-left corner
> * added code highlighting
> * changed tutorial heading levels as per request
> * updated the tutorial page to have simpler language as per request.
> 
> Note: The logo was something I was not able to enlarge it as the `spinxawesome-theme` does not have that feature. I have searched in the documentation and there is no mention on how to increase the size of the logo. If you know of the method to do it, please let me know and I can implement it.

@Roshan-Thomas Thanks a lot for looking into this. just a final request, would `furo` theme allow you to enlarge the logo? could you try with [this](https://github.com/TeamGraphix/graphix/blob/master/docs/logo/black_with_name.png) logo file?

**Comment by shinich1** [comment]
e.g. `quimb` docs has nicely-sized logo and they seem to use `furo` theme.

**Comment by Roshan-Thomas** [comment]
@shinich1 I have updated the theme to `furo` and got the logo (dark and light themed) to work. 

Also, I had to place the `dark` and `light` logo in the `_static` folder as `furo` reads the images from the `_static` folder only, and for some odd reason the file paths to the `logo` folder was not working. 

Hope this fixes all the issues and the changes required.

**Comment by shinich1** [comment]
> @shinich1 I have updated the theme to `furo` and got the logo (dark and light themed) to work.
> 
> Also, I had to place the `dark` and `light` logo in the `_static` folder as `furo` reads the images from the `_static` folder only, and for some odd reason the file paths to the `logo` folder was not working.
> 
> Hope this fixes all the issues and the changes required.

nice, it looks like the build failed - could you fix it?
https://app.readthedocs.org/projects/graphix-uh/

**Comment by shinich1** [comment]
> fixed the issue, the `furo` package was not present in the `requirements.txt`. I have added it now (see last commit).

will check once the build completed. I see lint CI failing below, could you `black -l 120` the `conf.py`?

**Comment by Roshan-Thomas** [comment]
> > fixed the issue, the `furo` package was not present in the `requirements.txt`. I have added it now (see last commit).
> 
> will check once the build completed. I see lint CI failing below, could you `black -l 120` the `conf.py` (see [our guide](https://github.com/TeamGraphix/graphix/blob/master/CONTRIBUTING.md))?

should I run the lint for any other files (other than `conf.py`)?

**Comment by shinich1** [comment]
@Roshan-Thomas all look good now - I will merge this once tests passed and I checked things again. 
btw, I'd appreciate if you could star the repo, if not done yet :)

**Comment by Roshan-Thomas** [comment]
> @Roshan-Thomas all look good now - I will merge this once tests passed and I checked things again. btw, I'd appreciate if you could star the repo, if not done yet :)

Thank you!

**Review by shinich1 — APPROVED** [review]

**Pull request merged**
